### PR TITLE
String literal syntax: support doubled-delimiter escape for double-quote (BT-743)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

Fixes BT-743: string literal syntax had no way to include a literal double-quote character.

Implements the **doubled-delimiter** convention: inside a double-quoted string, `""` (two adjacent double-quotes) produces a single literal `"` character. This follows standard Smalltalk/Newspeak convention.

**Before:** No way to embed `"` in a string — `\"` produced a 2-char string `\"`, and `""""` was a parse error.

**After:** `""""` is a 1-char string containing `"`, and `"say ""hello"" please"` produces `say "hello" please`.

## Key changes

- **`lex_string()`**: Refactored from span-based to buffer-based segment content accumulation, adding a `Some('"') if peek_char_n(1) == Some('"')` guard that advances past both quotes and pushes a single `"` into the content buffer
- **`skip_nested_string()`**: Same guard added for correct brace-depth tracking when a `""` appears inside a `{...}` interpolation expression
- **Tests**: 4 new unit tests (including doubled-delimiter + interpolation combination) and a new `tests/stdlib/string_quote_escape.bt` integration test
- **Docs**: Escape sequence table added to `docs/beamtalk-language-features.md`

## Linear issue

https://linear.app/beamtalk/issue/BT-743/string-literal-syntax-has-no-escape-for-double-quote-character

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Strings now support doubled delimiters ("" yields a literal ") and preserve backslash escapes inside strings.
  * Improved handling around interpolation boundaries so surrounding text and literals are captured correctly.

* **Documentation**
  * Added guidance and examples for string escape sequences and escaping braces.

* **Tests**
  * Added comprehensive tests covering doubled-delimiter behavior, escapes, and interpolation edge cases.

### CODE REVIEW

File: docs/beamtalk-language-features.md [hunk 1]
```diff
@@ -39,6 +39,13 @@ greeting := "Hello, 世界! 🌍"
 message := "Welcome, {name}!"
 emoji := "Status: {status} ✓"
 
+// Escape sequences inside strings:
+//   ""   doubled delimiter → literal double-quote character
+//   \{   escaped brace → literal {  (prevents interpolation)
+//   \}   escaped brace → literal }
```
[the remark this answers is not in the document]
Comment (jamesc) [reply docs/beamtalk-language-features.md]: Fixed in commit 4e77bc4 — updated  and  description to clarify backslash is preserved (→ literal /) and added a note about this being current lexer behavior.

File: tests/stdlib/string_quote_escape.bt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// String doubled-delimiter escape tests (BT-743)
+// Doubling the delimiter ("") inside a string produces a literal double-quote.
```
[the remark this answers is not in the document]
Comment (jamesc) [reply tests/stdlib/string_quote_escape.bt]: Fixed in commit 4e77bc4 — added the required Copyright and SPDX-License-Identifier header to the top of the file.
